ledger-core/virtual/integration/constructor_test.go

```diff
@@ -819,6 +820,171 @@ func TestVirtual_Constructor_PulseChangedWhileOutgoing(t *testing.T) {
+			assert.Equal(t, objectRef, request.Callee)
+			assert.Equal(t, outgoing, request.CallOutgoing)
+
+			msg := payload.VDelegatedCallResponse{Callee: request.Callee}
+
+			switch typedChecker.VDelegatedCallRequest.CountBefore() {
+			case 1:
+				assert.Zero(t, request.DelegationSpec)
+
+				firstExpectedToken = payload.CallDelegationToken{
+					TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+					PulseNumber:       server.GetPulse().PulseNumber,
+					Callee:            request.Callee,
+					Outgoing:          request.CallOutgoing,
+					DelegateTo:        server.JetCoordinatorMock.Me(),
+					Approver:          firstApprover,
+				}
+				msg.ResponseDelegationSpec = firstExpectedToken
+			case 2:
+				assert.Equal(t, firstExpectedToken, request.DelegationSpec)
+
+				secondExpectedToken = payload.CallDelegationToken{
+					TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+					PulseNumber:       server.GetPulse().PulseNumber,
+					Callee:            request.Callee,
+					Outgoing:          request.CallOutgoing,
+					DelegateTo:        server.JetCoordinatorMock.Me(),
+					Approver:          secondApprover,
+				}
+				msg.ResponseDelegationSpec = secondExpectedToken
+			default:
+				t.Fatal("unexpected")
+			}
+
+			server.SendPayload(ctx, &msg)
+			return false
+		})
+		typedChecker.VDelegatedRequestFinished.Set(func(finished *payload.VDelegatedRequestFinished) bool {
+			assert.Equal(t, objectRef, finished.Callee)
+			assert.Equal(t, secondExpectedToken, finished.DelegationSpec)
+			assert.Equal(t, []byte("state A"), finished.LatestState.State)
+			return false
+		})
+		typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+			assert.Equal(t, objectRef, res.Callee)
+			assert.Equal(t, []byte("finish A.New"), res.ReturnArguments)
+			assert.Equal(t, int(firstPulse), int(res.CallOutgoing.GetLocal().Pulse()))
+			assert.Equal(t, secondExpectedToken, res.DelegationSpec)
+			return false
+		})
+	}
+
+	pl := payload.VCallRequest{
+		CallType:       payload.CTConstructor,
+		CallFlags:      payload.BuildCallFlags(constructorIsolation.Interference, constructorIsolation.State),
+		Caller:         server.GlobalCaller(),
+		Callee:         classA,
+		CallSiteMethod: "New",
+		CallOutgoing:   outgoing,
+	}
+
+	server.SendPayload(ctx, &pl)
+
+	// wait for results
+	{
+		testutils.WaitSignalsTimed(t, 10*time.Second, synchronizeExecution.Wait())
+
+		tokenRequestDone := server.Journal.Wait(
+			predicate.ChainOf(
+				predicate.NewSMTypeFilter(&execute.SMDelegatedTokenRequest{}, predicate.AfterAnyStopOrError),
+				predicate.NewSMTypeFilter(&execute.SMExecute{}, predicate.BeforeStep((&execute.SMExecute{}).StepWaitExecutionResult)),
+			),
+		)
+		// wait for first pulse change
+		server.IncrementPulseAndWaitIdle(ctx)
+		testutils.WaitSignalsTimed(t, 20*time.Second, tokenRequestDone)
+
+		// wait for second pulse change
+		server.IncrementPulseAndWaitIdle(ctx)
+		testutils.WaitSignalsTimed(t, 20*time.Second, tokenRequestDone)
```

Comment:
Nope, once Wait is done it won't be checked second time. Please, check it using for loop and i < 2

Reply:
done